tests/behat/login_javascript.feature

```diff
@@ -0,0 +1,161 @@
+  I need to log in to the platform, having had my domain and credentials validated.
+
+  Background:
+    Given I log in as "admin"
+    And the following "users" exist:
+      | username | firstname | lastname | email                | password |
+      | student1 | Student   | 1        | student1@example.com | pass1    |
+      | student2 | Student   | 2        | student2@example.com | pass2    |
+    And the following "cohorts" exist:
+      | idnumber | name     |
+      | cht1     | Cohort 1 |
+      | cht2     | Cohort 2 |
+    And I add "student1@example.com" user to "Cohort 1" cohort members
+    And I add "student2@example.com" user to "Cohort 2" cohort members
+    And I visit the local URL "/local/brandmanager/manage_brand.php?"
+    And I set the following fields to these values:
+      | Name | Brand1 |
+    And I press "Save changes"
+    And I click on "Manage brand cohorts" "link" in the "Brand1" "table_row"
+    And I should see "Manage brand cohorts"
+    And I expand the "Selected Cohorts" autocomplete menu
+    And I click on "Cohort 1" item in the autocomplete menu
+    And I press "Save changes"
+    And I click on "Back to brands" "link"
+    And I click on "Manage domains" "link" in the "Brand1" "table_row"
+    And I should see "Add new domain"
+    And I set the following fields to these values:
+      | domain | http://192.168.120.50 |
+    And I press "Save changes"
+    And I click on "Back to brands" "link"
+    And I press "New brand"
+    And I set the following fields to these values:
+      | Name | Brand2 |
+    And I press "Save changes"
+    And I click on "Manage brand cohorts" "link" in the "Brand2" "table_row"
+    And I should see "Manage brand cohorts"
+    And I expand the "Selected Cohorts" autocomplete menu
+    And I click on "Cohort 2" item in the autocomplete menu
+    And I press "Save changes"
+    And I click on "Back to brands" "link"
+    And I click on "Manage domains" "link" in the "Brand2" "table_row"
+    And I should see "Add new domain"
+    And I set the following fields to these values:
+      | domain | otherdomain |
+    And I press "Save changes"
+    And I click on "Back to brands" "link"
+    And I log out
+    And I visit the local URL "/local/signin/index.php"
+
+  @javascript
+  Scenario: 01. Successful login, without redirect.
+    Given I should see "Username"
+    And I should not see "Password"
+    And I set the following fields to these values:
+      | username | student1 |
+    And I press "Proceed"
+    And I should see "Password"
+    And I set the following fields to these values:
+      | password | pass1 |
+    And I press "Log In"
+    And I should see "Dashboard"
+    And I should see "Site home"
+
+  @javascript
+  Scenario: 02. Redirect if on wrong domain.
+    Given I set the following fields to these values:
+      | username | student2 |
+    And I press "Proceed"
+    Then I should see "http://otherdomain/behat/local/signin/index.php?username=student2"
```

Comment:
Shouldn't we be checking the current URL here? This is checking for the presence of text on the page.